libs/plugin/src/executors/sync-repo/executor.ts

```diff
@@ -1,52 +0,0 @@
-import type { SyncRepoExecutorSchema } from './schema'
-import { workspaceRoot } from '@nx/devkit'
-import { execSync } from 'child_process'
-import * as fs from 'fs'
-import * as path from 'path'
-
-export default async function runExecutor(
-  options: SyncRepoExecutorSchema
-): Promise<{ success: boolean }> {
-  const {
-    outputPath,
-    repoUrl: url,
-    prefixPath: prefix = true,
-    branch,
-  } = options
-  const cwd = prefix ? path.join(workspaceRoot, outputPath) : outputPath
-  const remoteHash = getRemoteRepoHash(url)
-  const name = path.basename(cwd)
-
-  console.log(
-    `
-Caution: if this is part of nx cache replay,
-         no git command is actually executed.` + '\n '
-  )
-
-  if (fs.existsSync(cwd)) {
-    // Fetch & reset
-    const localHash = getLocalRepoHash(cwd)
-    if (remoteHash !== localHash) {
-      execSync(`git fetch --depth 1`, { cwd })
-      execSync(`git reset --hard ${branch}`, { cwd })
-    } else console.log('Repo already existed with the latest commit')
-  } else {
-    // Clone
-    const parent = path.dirname(cwd)
-    fs.mkdirSync(parent, { recursive: true })
-    execSync(`git clone ${url} --depth 1 ${name}`, { cwd: parent })
-  }
-
-  // Compute hash
-  const localHash = getLocalRepoHash(cwd)
-  console.log(`Synced ${name} with hash ${localHash}`)
-
-  const hashPath = path.format({ ...path.parse(cwd), base: '', ext: '.hash' })
-  fs.writeFileSync(hashPath, localHash)
```

Comment:
So I am not sure how submodules are updated, but basically its extremely expensive to store the entire dm repo in cache. This makes sure that we export the hash so that instead of checking for the entire repo, it only checks for a single hash file. 
in this case, how do we ensure that builds on the same submodule version is cached, and how do we ensure that builds are updated when submodule is updated?

Reply:
Theres conflicting documentation on whether NX natively supports submodules, so I added back the executor which should ensure it caches properly